### DISCUSSION TITLE
Oracle manipulation diagram

### DISCUSSION THREAD

**Discussion by fishy0x** [Cyfrin/security-and-auditing-full-course-s23#250] [Show and tell]
HI guys, i'm glad to share with you one of the diagrams that i made to help me understanding the oracle manipulation vulnerability on the thunderLoan protocol.

https://excalidraw.com/#json=5ppBbiXukTYGl-TXWMyiL,oIcZDfiA9fzBMKOnha0C9A

@PatrickAlphaC  if you want to add it to the diagram to the course resources, just do it.

**Comment by EngrPips** [comment]
Hello @JuanPeVentura, When I clicked on the link you shared, it did not show your diagram, but instead it showed the below

![Screenshot 2024-11-18 at 08 09 53](https://github.com/user-attachments/assets/e3dcb14d-7d38-482a-916d-71a136f3b1f6)

**Reply by fishy0x** [reply]
Here Is the png of the diamgram:

Oracle manipulación diagra.png](https://github.com/user-attachments/assets/678089e8-7854-45e0-b175-8485ed395edc)

**Reply by fishy0x** [reply]
Oh this Is strange, maybe the problem that re  you are having Is because the excaildraw Is trying to show you the tutorial, please use excaildraw, like put some text or whatever you want, then try to open the link again and click replace my content

**Reply by EngrPips** [reply]
I did what you asked me to do, and then I got this. 

![Screenshot 2024-11-18 at 12 27 17](https://github.com/user-attachments/assets/b09fbc4a-f16f-407d-9509-a4b6d2534685)

I am unsure why it is unable to load your image.

**Reply by fishy0x** [reply]
Here is the image: 

![387189788-678089e8-7854-45e0-b175-8485ed395edc](https://github.com/user-attachments/assets/756d1bfc-ea2f-4dda-be94-9bcc4dab8013)

**Reply by fishy0x** [reply]
You can try this link instead:

https://excalidraw.com/#json=8GsmsjgZZD4dRD_91TgXn,zidXbVD9aSxk8eWkw1yQQw

**Reply by fishy0x** [reply]
And here is the file of the diagram, you can just import it to excaildraw:

 https://file.io/FOBDKcFCuh5S

**Reply by fishy0x** [reply]
> I did what you asked me to do, and then I got this.
> 
> ![Screenshot 2024-11-18 at 12 27 17](https://private-user-images.githubusercontent.com/95760918/387196038-b09fbc4a-f16f-407d-9509-a4b6d2534685.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5Mzc4NDEsIm5iZiI6MTczMTkzNzU0MSwicGF0aCI6Ii85NTc2MDkxOC8zODcxOTYwMzgtYjA5ZmJjNGEtZjE2Zi00MDdkLTk1MDktYTRiNmQyNTM0Njg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE4VDEzNDU0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5OWI5MzY4ODlmOWY5ODQwNzgxNmExODYxZDUwZjA0ZTZhOTExY2QyYTA4NGZkMjA1NDg1MWUyODdjNDVkODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.r--VKj5xdSHqUIa1U_DEpqQ-hoFa-KhXEDEIZ04m5hM)
> 
> I am unsure why it is unable to load your image.

Bro,  i want your search bar organization, how did you do that ??

**Reply by EngrPips** [reply]
> > I did what you asked me to do, and then I got this.
> > ![Screenshot 2024-11-18 at 12 27 17](https://private-user-images.githubusercontent.com/95760918/387196038-b09fbc4a-f16f-407d-9509-a4b6d2534685.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE5Mzc4NDEsIm5iZiI6MTczMTkzNzU0MSwicGF0aCI6Ii85NTc2MDkxOC8zODcxOTYwMzgtYjA5ZmJjNGEtZjE2Zi00MDdkLTk1MDktYTRiNmQyNTM0Njg1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTE4VDEzNDU0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5OWI5MzY4ODlmOWY5ODQwNzgxNmExODYxZDUwZjA0ZTZhOTExY2QyYTA4NGZkMjA1NDg1MWUyODdjNDVkODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.r--VKj5xdSHqUIa1U_DEpqQ-hoFa-KhXEDEIZ04m5hM)
> > I am unsure why it is unable to load your image.
> 
> Bro, i want your search bar organization, how did you do that ??

If you update Chrome browser to its latest version, you will have access to that functionality 

![Screenshot 2024-11-18 at 14 54 55](https://github.com/user-attachments/assets/61163c2b-3f52-491d-bce9-f5b8854b3a4f)

**Reply by EngrPips** [reply]
> And here is the file of the diagram, you can just import it to excaildraw:
> 
> https://file.io/FOBDKcFCuh5S

Thanks buddy.
